### PR TITLE
state transition diagram

### PR DESCRIPTION
**Pull request by trgill**
PoolSpace.gv - Physical space states for a pool
PoolHealth.gv -  Pool Health (thin_check) and logical space states
PoolAssembly.gv - Pool assembly states (labels, members, metadata)
Filesystem.gv - Filesystem health (xfs_check) and logical space states

Note: These diagrams don't attempt to deal with the timing issues the code will need to address.  For example, as soon as we check that a pool member is available - it could then go away.

### CODE REVIEW

**Comment by agrover** [comment]
Please omit .ps files from commit. Consider adding *.ps to .gitignore?

File: docs/design/Makefile [hunk 1]
```diff
@@ -1,8 +1,16 @@
 StratisSoftwareDesign.pdf: StratisSoftwareDesign.lyx
 	lyx -batch --export pdf $<
 
+PoolStateDiagrams:
```
Comment (mulkieran) [inline docs/design/Makefile]: I think that svg might be a better output file type.

File: docs/design/Makefile [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+	dot -Tps PoolAssembly.gv -o PoolAssembly.ps
+	dot -Tps PoolHealth.gv -o PoolHealth.ps
+	dot -Tps PoolSpace.gv -o PoolSpace.ps
+	dot -Tps Filesystem.gv -o Filesystem.ps
+
 clean:
 	-rm StratisSoftwareDesign.pdf
+	-rm PoolStateDiagram.ps
```
Comment (mulkieran) [inline docs/design/Makefile]: Should remove all output files.

File: docs/design/Makefile (hunk 1 above)
Comment (trgill) [reply docs/design/Makefile]: Thanks.  Switched to .svg.

**Pull request closed without merging**